src/ORM/DataObjectSchema.php

```diff
@@ -201,6 +201,11 @@ public function fieldSpecs($classOrInstance, $options = 0)
         $db = [];
         $classes = $uninherited ? [$class] : ClassInfo::ancestry($class);
         foreach ($classes as $tableClass) {
+            // Skip irrelevant parent classes
+            if ($tableClass !== DataObject::class && !is_subclass_of($tableClass, DataObject::class)) {
```

Comment:
@tractorcow Omitting the `$tableClass !== DataObject::class` saves around another 500 method calls. As far as I can tell `$this->databaseFields(DataObject::class);` doesn’t return anything, do you think that would be safe?

Reply:
Nice job. :)